### PR TITLE
fix(cli): use global_state file_fetcher when using SpecifierHandler

### PR DESCRIPTION
**Pull request by kitsonk**
Fixes: #7709

### CODE REVIEW

File: cli/specifier_handler.rs [hunk 1]
```diff
@@ -204,27 +204,16 @@ pub struct FetchHandler {
...
-      ca_file.as_deref(),
-    )?;
     let disk_cache = deno_dir.gen_cache;
+    let file_fetcher = global_state.file_fetcher.clone();
 
     Ok(FetchHandler {
       disk_cache,
       file_fetcher,
-      permissions: permissions.clone(),
+      permissions: Permissions::allow_all(),
```
Comment (bartlomieju) [inline cli/specifier_handler.rs:216]: Careful - this might be permission escape - `permissions` should still be passed down from `GlobalState`

**Review by bartlomieju — COMMENTED** [review]
Just need a small change, otherwise LGTM

File: cli/specifier_handler.rs (hunk 1 above)
Comment (kitsonk) [reply cli/specifier_handler.rs:216]: I thought I tried that and it broke things, like required you to use `--allow-read` to even read a local module. I will try again though.

Comment (bartlomieju) [reply cli/specifier_handler.rs:216]: This constructor is called during `prepare_module_load` - so the permissions passed from `prepare_module_load` will definitely be the proper ones.

Comment (kitsonk) [reply cli/specifier_handler.rs:216]: Ah, I think that was the problem, I was cloning them off of the `global_state`... 🤦

**Review by bartlomieju — APPROVED** [review]
Lgtm

**Pull request merged**